### PR TITLE
ghcjs: fix build and disable profiling by default

### PR DESCRIPTION
**Pull request by yorickvP**
This updates ghcjs so it builds, but profiling is currently broken, https://github.com/ghcjs/ghcjs/issues/759

cc @peti @basvandijk  @ElvishJerricco  @vaibhavsagar

### CODE REVIEW

**Review by vaibhavsagar — APPROVED** [review]

File: pkgs/development/haskell-modules/generic-builder.nix [hunk 1]
```diff
@@ -25,7 +25,7 @@ in
 , doBenchmark ? false
 , doHoogle ? true
 , editedCabalFile ? null
-, enableLibraryProfiling ? true
+, enableLibraryProfiling ? !(ghc.isGhcjs or false)
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/generic-builder.nix]: Could this just be `!ghc.isGhcjs`?

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 2]
```diff
@@ -105979,6 +105979,26 @@ self: {
        hydraPlatforms = stdenv.lib.platforms.none;
      }) {};
 
+  "happy_1_19_9" = callPackage
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/hackage-packages.nix]: This file shouldn't be updated in PRs.  It is updated automatically:

https://github.com/NixOS/nixpkgs/pull/67438#issuecomment-525522019

**Review by cdepillabout — APPROVED** [review]
Other than the two above comments, this LGTM.  I tried building this locally and it looks like it built correctly.  I tried compiling a small module and it appeared to work.

**Comment by cdepillabout** [comment]
I'm not sure if we want to reduce the number of generated Haskell packages, but this also works with the older version of `happy` we already have:

```diff
diff --git a/pkgs/development/compilers/ghcjs-ng/default.nix b/pkgs/development/compilers/ghcjs-ng/default.nix
index 6c39263ca3d..5608fefc54b 100644
--- a/pkgs/development/compilers/ghcjs-ng/default.nix
+++ b/pkgs/development/compilers/ghcjs-ng/default.nix
@@ -24,7 +24,7 @@ let
     configuredSrc = callPackage ./configured-ghcjs-src.nix {
       inherit ghcjsSrc;
       inherit (bootPkgs) ghc alex;
-      happy = bootPkgs.happy_1_19_9;
+      happy = bootPkgs.happy_1_19_5;
     };
     genStage0 = callPackage ./mk-stage0.nix { inherit (passthru) configuredSrc; };
     bootPkgs = bootPkgs.extend (lib.foldr lib.composeExtensions (_:_:{}) [
@@ -36,7 +36,7 @@ let
       (callPackage ./common-overrides.nix {
         inherit haskellLib;
         inherit (bootPkgs) alex;
-        happy = bootPkgs.happy_1_19_9;
+        happy = bootPkgs.happy_1_19_5;
       })
       ghcjsDepOverrides
     ]);
```

**Review by peti — APPROVED** [review]
Looks good to me.

**Comment by ElvishJerricco** [comment]
LGTM

**Review by peti — CHANGES_REQUESTED** [review]
Don't edit `hackage-packages.nix`.

**Comment by yorickvP** [comment]
@peti okay, updated

**Comment by yorickvP** [comment]
@peti is it possible to get this into 19.09?

**Comment by vaibhavsagar** [comment]
@yorickvP you can make a PR against the `release-19.09` branch yourself.

**Pull request merged**